gufe/solventcomponent.py

```diff
@@ -145,3 +151,13 @@ def to_dict(self):
                 'negative_ion': self.negative_ion,
                 'ion_concentration': ion_conc,
                 'neutralize': self._neutralize}
+
+    def to_storage_ready(self):
+        default_keys = get_defaults_from_init(self, self.to_dict())
+        return generic_to_storage_ready(self, self._storage_path,
+                                        default_keys)
+
+    @classmethod
+    def from_storage_bytes(cls, serialized_bytes, load_func):
```

Comment:
Is it necessary for `load_func` to be a parameter here? From the tests it looks like it should be a detail of the `from_storage_bytes` method implemented for a given component. It's really not clear to me externally what it should be for a given component, and I think `from_storage_bytes` should only require `serialized_bytes` to do its job.

Reply:
Example -- Say I have this:

```python
class A:
    foo: str

class B:
    a: A
```

Let's assume that we also have the relevant serialization stuff. We serialize these objects separately (maybe instances of `A` are frequently reused), so when we serialize an instance of `B` we get 2 JSON files:

myroot/HASH_FOR_B.json

```json
{ "a": {":module:": "__main__",
        ":class:": "A",
        ":path:": "HASH_FOR_A.json"}
}
```

myroot/HASH_FOR_A.json

```json
{"foo": "user input"}
```

So here's why we need more than just the `serialized_bytes`: when you try to deserialize the instance of `B`, `serialized_bytes` is only the contents of `HASH_FOR_B`. The value of the `A`'s `foo` is not included. You need to know where to go to look up a label (`:path:`) -- and that's what the `load_func` does. It takes a label and returns that information. `load_func` will typically be an instance method of the storage object.